pkg/manifests/external_dns.go

```diff
@@ -143,53 +187,90 @@ func newExternalDNSClusterRole(conf *config.Config, externalDnsConfig *ExternalD
 				Resources: []string{"endpoints", "pods", "services", "configmaps"},
 				Verbs:     []string{"get", "watch", "list"},
 			},
-			{
-				APIGroups: []string{"extensions", "networking.k8s.io"},
-				Resources: []string{"ingresses"},
-				Verbs:     []string{"get", "watch", "list"},
-			},
 			{
 				APIGroups: []string{""},
 				Resources: []string{"nodes"},
 				Verbs:     []string{"get", "watch", "list"},
 			},
 		},
 	}
+
+	switch externalDnsConfig.ResourceType {
```

Comment:
why not make this logic a method on your new config type?

Reply:
you mean make it a method on the resource type?

I think it makes more sense to be left as a separate function - the rbac isn't as much a property or action of the externaldns config as it is a set of resources that need to be generated according to state in it (hence function vs method). If you feel strongly I can refactor, but doesn't seem necessary or flexible, especially if roles need to change based on other objects/parameters in the future

Reply:
I see now what you mean, took this out to a separate method (addResourceSpecificRules)